qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -52,12 +52,16 @@ class QubitConverter:
     def __init__(self,
                  mapper: QubitMapper,
                  two_qubit_reduction: bool = False,
+                 auto_symmetry_reduction: bool = False,
                  z2symmetry_reduction: Optional[List[int]] = None):
         """
 
         Args:
             mapper: A mapper instance used to convert second quantized to qubit operators
             two_qubit_reduction: Whether to carry out two qubit reduction when possible
+            auto_symmetry_reduction: Whether to perform automatic symmetry reduction. This requires
+                the `sector_locator` argument of `convert` to be specified. This option will also be
+                overwritten if `z2symmetry_reduction` is not `None`.
             z2symmetry_reduction: An optional sector definition so the desired operator, from
                the tapered set, containing the problem solution is returned. This is a list of
                -1 and 1's to define the sector, where the list size is the number of symmetries
```

Comment:
If we are specifying the auto reduction here maybe we should just have it like before where z2_symmetry reduction was either None, a manually given sector (List[int]) or the 'auto' string. Evidently auto will only work with problems/algorithms that supply a the sector locator to convert. By having auto, manual or none in the same parameter I think its clearer to an end user.

As it stands now, the one from sector locator would override any manual one (as long as it returns a non-None value back which is what I expected). Had it returned None. With both on the one param, as they were before, we could never get into that situation since it would no symmetry, manual or auto.